test_tag_order.py

```diff
@@ -0,0 +1,163 @@
+    )
+    return (model_result, track)
+
+
+def test_model_heirechy_wallabies():
+    config = test_config()
+    original_result = model_result(config.models[0], "wallaby")
+    retrained_result = model_result(config.models[1], "wallaby")
+    resnet_result = model_result(config.models[2], "wallaby")
+    wallaby_result = model_result(config.models[3], None)
+    wallaby_old_result = model_result(config.models[4], "wallaby")
+    results = [
+        original_result,
+        retrained_result,
+        resnet_result,
+        wallaby_result,
+        wallaby_old_result,
+    ]
+    # old wallaby is chosen over no tag
+    master_tag = thermal.get_master_tag(results, wallaby_device=True)
+    assert master_tag[0].model_config.name == "wallaby-old"
+    assert master_tag[1]["tag"] == "wallaby"
+
+    # new wallaby tag is chosen over old
+    wallaby_result[1]["tag"] = "wallaby"
+    master_tag = thermal.get_master_tag(results, wallaby_device=True)
+    assert master_tag[0].model_config.name == "wallaby"
+    assert master_tag[1]["tag"] == "wallaby"
+
+    wallaby_result[1]["tag"] = "bird"
+    wallaby_old_result[1]["tag"] = "possum"
+
+    # no wallaby tags, means no tag
+    master_tag = thermal.get_master_tag(results, wallaby_device=True)
+    assert master_tag is None
+
+
+def test_model_heirechy():
+    config = test_config()
+    original_result = model_result(config.models[0], "bird")
+    retrained_result = model_result(config.models[1], "cat")
+    resnet_result = model_result(config.models[2], "possum")
+    wallaby_result = model_result(config.models[3], "not")
+    wallaby_old_result = model_result(config.models[4], "not")
+    results = [
+        original_result,
+        retrained_result,
+        resnet_result,
+        wallaby_result,
+        wallaby_old_result,
+    ]
+
+    # original bird classification overrules all others
+    master_tag = thermal.get_master_tag(results, wallaby_device=False)
+    assert master_tag[0].model_config.name == "original"
+    assert master_tag[1]["tag"] == "bird"
+
+    # if the original model isn't a bird, resnet is the next best
+    original_result[1]["tag"] = "cat"
+    master_tag = thermal.get_master_tag(results, wallaby_device=False)
+    assert master_tag[0].model_config.name == "resnet"
+    assert master_tag[1]["tag"] == "possum"
+
+    # if resent doens't know, use retrained
+    resnet_result[1]["tag"] = None
+    master_tag = thermal.get_master_tag(results, wallaby_device=False)
+    assert master_tag[0].model_config.name == "retrained"
+    assert master_tag[1]["tag"] == "cat"
+
+    # if resent is unidentified use retrained
+    resnet_result[1]["tag"] = "unidentified"
+    master_tag = thermal.get_master_tag(results, wallaby_device=False)
+    assert master_tag[0].model_config.name == "retrained"
+    assert master_tag[1]["tag"] == "cat"
+
+    # if all models are unidentified use unidentified
+    retrained_result[1]["tag"] = "unidentified"
```

Comment:
You have tested that all unidentified models returns unidentfied but not that only one unidentified returns unidentified.